File: Components/WinForms/VB/EditorsMainDemo/Modules/ModuleTimeSpanEdit.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository

Namespace DevExpress.XtraEditors.Demos
	Partial Public Class ModuleTimeSpanEdit
		Inherits TutorialControl
		Private ReadOnly Property TimeSpanEdit() As TimeSpanEdit
			Get
				Return timeSpanEdit_Renamed
			End Get
		End Property
		Public Sub New()
			InitializeComponent()
            TutorialInfo.WhatsThisCodeFile = "VB\EditorsMainDemo\Modules\ModuleTimeSpanEdit.vb"
            TutorialInfo.WhatsThisXMLFile = "timespanedit.xml"
		End Sub
		Private Sub OnModuleLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			SetDisplayFormat()
			ShowValue()
		End Sub
		Private Sub OnTimeSpanEditEditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles timeSpanEdit_Renamed.EditValueChanged
			ShowValue()
		End Sub
		Private Sub ShowValue()
            Dim result As String = String.Empty
            If TimeSpanEdit.EditValue IsNot Nothing Then
                result = TimeSpanEdit.EditValue.ToString()
            End If
            lbValue.Text = result
        End Sub
		' <cbEditFormat>
		Private Sub OnEditFormatChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbEditFormat.SelectedIndexChanged
			SetEditFormat()
		End Sub
		Private Sub SetEditFormat()
			Dim formats() As String = { "", "dd.HH:mm:ss", "dd.HH:mm", "HH:mm:ss", "HH:mm", "mm:ss", "dd", "HH", "mm", "ss" }
			Dim index As Integer = cbEditFormat.SelectedIndex
			TimeSpanEdit.Properties.Mask.MaskType = Mask.MaskType.Custom
			TimeSpanEdit.Properties.Mask.EditMask = formats(index)
            teEditFormatString.Text = formats(index)
		End Sub
		' </cbEditFormat>
		' <cbDisplayFormat>
		Private Sub OnDisplayFormatChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbDisplayFormat.SelectedIndexChanged
			SetDisplayFormat()
		End Sub
		Private Sub SetDisplayFormat()
			Dim formats() As String = { "", "{0:dd}.{0:hh}:{0:mm}:{0:ss}", "{0:dd}.{0:hh}:{0:mm}", "{0:hh}:{0:mm}:{0:ss}", "{0:hh}:{0:mm}", "{0:mm}:{0:ss}", "{0:dd} days", "{0:hh} hours", "{0:mm} minutes", "{0:ss} seconds" }
			Dim index As Integer = cbDisplayFormat.SelectedIndex
			TimeSpanEdit.Properties.DisplayFormat.FormatType = Utils.FormatType.Custom
			TimeSpanEdit.Properties.DisplayFormat.FormatString = formats(index)
            teDisplayFormatString.Text = formats(index)
		End Sub
		' </cbDisplayFormat>
		' <seMaxDays>
		Private Sub OnMaxDaysValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles seMaxDays.EditValueChanged
			TimeSpanEdit.Properties.MaxDays = CInt(Fix(seMaxDays.Value))
		End Sub
		' </seMaxDays>
		' <seMaxHours>
		Private Sub OnMaxHoursValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles seMaxHours.EditValueChanged
			TimeSpanEdit.Properties.MaxHours = CInt(Fix(seMaxHours.Value))
		End Sub
		' </seMaxHours>
		' <seMaxMinutes>
		Private Sub OnMaxMinutesValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles seMaxMinutes.EditValueChanged
			TimeSpanEdit.Properties.MaxMinutes = CInt(Fix(seMaxMinutes.Value))
		End Sub
		' </seMaxMinutes>
		' <seMaxSeconds>
		Private Sub OnMaxSecondsValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles seMaxSeconds.EditValueChanged
			TimeSpanEdit.Properties.MaxSeconds = CInt(Fix(seMaxSeconds.Value))
		End Sub
		' </seMaxSeconds>
		' <ceAllowEditDays>
		Private Sub OnAllowEditDaysChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ceAllowEditDays.CheckedChanged
			TimeSpanEdit.Properties.AllowEditDays = Not TimeSpanEdit.Properties.AllowEditDays
		End Sub
		' </ceAllowEditDays>
		' <ceAllowEditHours>
		Private Sub OnAllowEditHoursChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ceAllowEditHours.CheckedChanged
			TimeSpanEdit.Properties.AllowEditHours = Not TimeSpanEdit.Properties.AllowEditHours
		End Sub
		' </ceAllowEditHours>
		' <ceAllowEditMinutes>
		Private Sub OnAllowEditMinutesChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ceAllowEditMinutes.CheckedChanged
			TimeSpanEdit.Properties.AllowEditMinutes = Not TimeSpanEdit.Properties.AllowEditMinutes
		End Sub
		' </ceAllowEditMinutes>
		' <ceAllowEditSeconds>
		Private Sub OnAllowEditSecondsChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ceAllowEditSeconds.CheckedChanged
			TimeSpanEdit.Properties.AllowEditSeconds = Not TimeSpanEdit.Properties.AllowEditSeconds
		End Sub
		' </ceAllowEditSeconds>
		' <ceReadOnly>
		Private Sub OnReadOnlyChanged(ByVal sender As Object, ByVal e As EventArgs) Handles ceReadOnly.CheckedChanged
			TimeSpanEdit.Properties.ReadOnly = Not TimeSpanEdit.Properties.ReadOnly
			cbEditFormat.Enabled = Not cbEditFormat.Enabled
		End Sub
		' </ceReadOnly>
	End Class
End Namespace

File: Components/WinForms/CS/EditorsMainDemo/Modules/ModuleTimeSpanEdit.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;

namespace DevExpress.XtraEditors.Demos {
    public partial class ModuleTimeSpanEdit: TutorialControl {
        TimeSpanEdit TimeSpanEdit { get { return timeSpanEdit; } }
        public ModuleTimeSpanEdit() {
            InitializeComponent();
            TutorialInfo.WhatsThisCodeFile = "CS\\EditorsMainDemo\\Modules\\ModuleTimeSpanEdit.cs";
            TutorialInfo.WhatsThisXMLFile = "DevExpress.XtraEditors.Demos.CodeInfo.timespanedit.xml";
        }
        protected override void UpdateTabControlPosition(XtraTab.XtraTabControl tabControl) { }
        protected override void CalcContentCore() {
            CalcLayoutControlBounds(layoutControl1);
        }
        void OnModuleLoad(object sender, EventArgs e) {
            SetDisplayFormat();
            ShowValue();
        }
        void OnTimeSpanEditEditValueChanged(object sender, EventArgs e) {
            ShowValue();
        }
        void ShowValue() {
            string result = string.Empty;
            if(TimeSpanEdit.EditValue != null)
                result = TimeSpanEdit.EditValue.ToString();
            lbValue.Text = result;
        }
        // <cbEditFormat>
        void OnEditFormatChanged(object sender, EventArgs e) {
            SetEditFormat();
        }
        void SetEditFormat() {
            string[] formats = new string[] { 
                "",
                "dd.HH:mm:ss",
                "dd.HH:mm",
                "HH:mm:ss",
                "HH:mm", 
                "mm:ss", 
                "dd", 
                "HH", 
                "mm", 
                "ss" 
            };
            int index = cbEditFormat.SelectedIndex;
            TimeSpanEdit.Properties.Mask.MaskType = Mask.MaskType.Custom;
            teEditFormatString.Text = TimeSpanEdit.Properties.Mask.EditMask = formats[index];
        }
        // </cbEditFormat>
        // <cbDisplayFormat>
        void OnDisplayFormatChanged(object sender, EventArgs e) {
            SetDisplayFormat();
        }
        void SetDisplayFormat() {
            string[] formats = new string[] { 
                "",
                "{0:dd}.{0:hh}:{0:mm}:{0:ss}",
                "{0:dd}.{0:hh}:{0:mm}", 
                "{0:hh}:{0:mm}:{0:ss}",
                "{0:hh}:{0:mm}", 
                "{0:mm}:{0:ss}", 
                "{0:dd} days", 
                "{0:hh} hours", 
                "{0:mm} minutes", 
                "{0:ss} seconds" 
            };
            int index = cbDisplayFormat.SelectedIndex;
            TimeSpanEdit.Properties.DisplayFormat.FormatType = Utils.FormatType.Custom;
            teDisplayFormatString.Text = TimeSpanEdit.Properties.DisplayFormat.FormatString = formats[index];
        }
        // </cbDisplayFormat>
        // <seMaxDays>
        void OnMaxDaysValueChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.MaxDays = (int)seMaxDays.Value;
        }
        // </seMaxDays>
        // <seMaxHours>
        void OnMaxHoursValueChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.MaxHours = (int)seMaxHours.Value;
        }
        // </seMaxHours>
        // <seMaxMinutes>
        void OnMaxMinutesValueChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.MaxMinutes = (int)seMaxMinutes.Value;
        }
        // </seMaxMinutes>
        // <seMaxSeconds>
        void OnMaxSecondsValueChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.MaxSeconds = (int)seMaxSeconds.Value;
        }
        // </seMaxSeconds>
        // <ceAllowEditDays>
        void OnAllowEditDaysChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.AllowEditDays = !TimeSpanEdit.Properties.AllowEditDays;
        }
        // </ceAllowEditDays>
        // <ceAllowEditHours>
        void OnAllowEditHoursChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.AllowEditHours = !TimeSpanEdit.Properties.AllowEditHours;
        }
        // </ceAllowEditHours>
        // <ceAllowEditMinutes>
        void OnAllowEditMinutesChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.AllowEditMinutes = !TimeSpanEdit.Properties.AllowEditMinutes;
        }
        // </ceAllowEditMinutes>
        // <ceAllowEditSeconds>
        void OnAllowEditSecondsChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.AllowEditSeconds = !TimeSpanEdit.Properties.AllowEditSeconds;
        }
        // </ceAllowEditSeconds>
        // <ceReadOnly>
        void OnReadOnlyChanged(object sender, EventArgs e) {
            TimeSpanEdit.Properties.ReadOnly = !TimeSpanEdit.Properties.ReadOnly;
            cbEditFormat.Enabled = !cbEditFormat.Enabled;
        }
        // </ceReadOnly>
    }
}
